### PR TITLE
Weibull distribution implemented

### PR DESCRIPTION
**Pull request by mossadal**
Implementation of Weibull distribution (keeping the php 5.2 compatibility with actual code in /ugly)

### CODE REVIEW

**Comment by mossadal** [comment]
Looks like the random number generator is different in various versions of PHP, and that I set the tolerance a bit too tight in the sampling tests.

**Comment by gburtini** [comment]
LGTM other than the failing tests. I will modify your acceptance range, but I don't want to merge till you confirm that is a correct decision.

File: tests/WeibullDistributionTest.php [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+		$this->assertEquals($number, $mean, "Attempting to draw from Weibull(3.2, 1.4)) {$scale} times gives us a value too far from the expected mean. This could be just random chance.", 0.001);
+
+		$p = $counter / $scale;
+		$this->assertEquals($p, 1 - $d->cdf($cutoff), "Attempting to draw from  Weibull(3.2, 1.4))  {$scale} times gives the wrong number of values greater than {$cutoff}. This could be just random chance.", 0.001);
+	}
+
+
+
+	public function test_Weibull_PDF() {
+		$accuracy = 1e-10;
```
Comment (gburtini) [inline tests/WeibullDistributionTest.php]: Why are the expected accuracies different in different cases? Should this be a global constant?

File: tests/WeibullDistributionTest.php [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+		$d = new Weibull(0.2, 3.0);
+		$this->assertEquals(3.25296000000e7, $d->variance(), "variance incorrect", 1e-6);
+	}
+
+	public function test_Weibull_CDF_ICDF() {
+		$accuracy = 1e-8;
+
+		for($i = 0; $i < 100; $i++) {
+			$k = 3 * mt_rand(0, mt_getrandmax() - 1) / mt_getrandmax();
```
Comment (gburtini) [inline tests/WeibullDistributionTest.php]: What are these numbers?

Comment (gburtini) [reply tests/WeibullDistributionTest.php]: (perhaps leave a comment explaining them)

File: tests/WeibullDistributionTest.php (hunk 1 above)
Comment (mossadal) [reply tests/WeibullDistributionTest.php]: I guess we should strive for close to machine epsilon accuracy. I'll have a look at all the tests for this.

**Comment by mossadal** [comment]
Please wait with merging, I'll have a look at the above first.

**Comment by gburtini** [comment]
I weakly confirmed the RNG change; `mt_rand` got new seed mappings and `rand` now is `mt_rand` in 7.1+.

https://wiki.php.net/rfc/rng_fixes

I'm now working to get this merged. I'm hacking around the issue by making the tests less tight right now.

**Pull request merged**